libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/couchdb/lucene/LuceneAwareDatabaseConnector.java

```diff
@@ -183,21 +184,24 @@ public void setResultLimit(int limit) {
         return projectList.stream().filter(ProjectPermissions.isVisible(user)).collect(Collectors.toList());
     }
 
-    private static String formatSubquery(Set<String> filterSet, final String fieldname) {
+    private static String formatSubquery(Set<String> filterSet, final String fieldName) {
         final Function<String, String> addType = input -> {
-            if (fieldname.equals("state")) {
-                return fieldname + ":\"" + (enumByString(input, ProjectState.class).toString()) + "\"";
-            } else if (fieldname.equals("projectType")) {
-                return fieldname + ":\"" + (enumByString(input, ProjectType.class).toString()) + "\"";
+            if (fieldName.equals("state")) {
+                return fieldName + ":\"" + (enumByString(input, ProjectState.class).toString()) + "\"";
+            } else if (fieldName.equals("projectType")) {
+                return fieldName + ":\"" + (enumByString(input, ProjectType.class).toString()) + "\"";
             } else if (input.contains(" ")){
-                return fieldname + ":\"" + input + "\"";
+                return fieldName + ":\"" + input + "\"";
             } else {
-                return fieldname + ":" + input;
+                return fieldName + ":" + input;
             }
         };
 
         FluentIterable<String> searchFilters = FluentIterable.from(filterSet).transform(addType);
         return "( " + OR.join(searchFilters) + " ) ";
     }
 
+    public static String sanitizeQueryInput(String input) {
+        return nullToEmpty(input).replaceAll("[^a-zA-Z0-9]", " ").trim();
```

Comment:
This will erase umlauts and accented characters from search input. I don't think that's desired.

See [stackoverflow](https://stackoverflow.com/questions/1611979/remove-all-non-word-characters-from-a-string-in-java-leaving-accented-charact) for a possible solution.

Reply:
changed replaceAll parameters to
`replaceAll("[^\\p{L}\\p{Nd}]+"`

Reply:
I think it will do good to add a short comment explaining the regex

Reply:
added comment